src/coprocessor/codec/convert.rs

```diff
@@ -375,6 +381,54 @@ pub fn convert_json_to_uint(ctx: &mut EvalContext, json: &Json, tp: FieldTypeTp)
     convert_uint_to_uint(ctx, val, tp)
 }
 
+/// Converts a `DateTime` to printable string representation
+#[inline]
+pub fn convert_datetime_to_numeric_string(dt: &DateTime) -> String {
+    let time = dt.as_ref();
+    if dt.get_time_type() == TimeType::Date {
+        if dt.is_zero() {
+            String::from("00000000")
+        } else {
+            format!("{:04}{:02}{:02}", time.year(), time.month(), time.day())
+        }
+    } else {
+        let s = if dt.is_zero() {
+            String::from("00000000000000")
+        } else {
+            format!("{}", time.format("%Y%m%d%H%M%S"))
+        };
+        let fsp = dt.get_fsp();
+        if fsp > 0 {
+            // Do we need to round the result?
+            let nanos = time.nanosecond() / TEN_POW[9 - fsp as usize];
+            format!("{}.{1:02$}", s, nanos, fsp as usize)
+        } else {
+            s
+        }
+    }
+}
+
+/// Converts a `Duration` to printable numberic string representation
+#[inline]
+pub fn convert_duration_to_numeric_string(dur: Duration) -> String {
+    let mut buf = String::with_capacity(13);
+    if dur.neg() {
+        buf.push_str("-");
+    }
+    buf.push_str(&format!(
+        "{:02}{:02}{:02}",
+        dur.hours(),
+        dur.minutes(),
+        dur.secs(),
+    ));
+    let fsp = dur.fsp();
+    if fsp > 0 {
+        let nanos = dur.subsec_micros() / (10u32.pow(MICRO_WIDTH as u32 - u32::from(fsp)));
+        buf.push_str(&format!(".{:01$}", nanos, fsp as usize));
+    }
+    buf
+}
```

Comment:
use `write!` seems faster than `String;:push_str(format!(xxxx))`(because it can ).

```rust
#[cfg(test)]
mod bench_format {
    use super::test;
    use std::fmt::Write;

    #[bench]
    fn bench_format(b: &mut test::Bencher) {
        let mut s = String::from("abcde");
        let mut idx = 0;
        b.iter(|| {
            s.push_str(&format!("{}", idx));
            idx += 1;
        });
        println!("{:#?}", s);
    }

    #[bench]
    fn bench_write(b: &mut test::Bencher) {
        let mut s = String::from("abcde");
        let mut idx = 0;
        b.iter(|| {
            write!(s, "{}", idx);
            idx += 1;
        });
        println!("{:#?}", s);
    }
}
// running 2 tests
// test benchmark::bench_format::bench_format          ... bench:          47 ns/iter (+/- 4)
// test benchmark::bench_format::bench_write           ... bench:          25 ns/iter (+/- 1)
```